lib/search.rb

```diff
@@ -1447,7 +1451,7 @@ def topic_search
 
   def posts_eager_loads(query)
     query = query.includes(:user, :post_search_data)
-    topic_eager_loads = [:category]
+    topic_eager_loads = [{ category: :parent_category }]
```

Comment:
Just confirming here that there isn't any problems here even if category nesting is 3 because we only always display a single parent.